helix-term/src/commands/typed.rs

```diff
@@ -1370,13 +1370,38 @@ fn sort_reverse(
         return Ok(());
     }
 
-    sort_impl(cx, args, true)
+    sort_impl(cx, args, true, false)
+}
+
+fn sort_numeric(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    sort_impl(cx, args, false, true)
+}
+
+fn sort_numeric_reverse(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    sort_impl(cx, args, true, true)
 }
 
 fn sort_impl(
     cx: &mut compositor::Context,
     _args: &[Cow<str>],
     reverse: bool,
+    alphanumeric: bool,
 ) -> anyhow::Result<()> {
```

Comment:
As a suggestion you remove both `reverse` and `alphanumeric` and pass a single arguments called `cmp_fn` which is a `FnMut(&Tendril, &Tendril) -> Ordering` leaving to the caller to say how it want the ordering.

See: https://github.com/helix-editor/helix/pull/8436